### PR TITLE
UTF-8 test fix for 18.3

### PR DESCRIPTION
**Pull request by lukebakken**
Fix test for 18.3

### CODE REVIEW

**Comment by lukebakken** [comment]
@zeeshanlakhani adding `/utf8` to the binary literal fixed the issue.

File: tools.mk [hunk 1]
```diff
@@ -95,8 +130,8 @@ dialyzer-run:
 		| grep -F -f dialyzer.ignore-warnings.tmp -v \
 		| sed -E 's/^[[:space:]]*[0-9]+[[:space:]]*//' \
 		| sed -E 's/([]\^:+?|()*.$${}\[])/\\\1/g' \
-		| sed -E 's/(\\\.erl\\\:)/\1\\d+:/g' \
-		| sed -E 's/^(.*)$$/^\1$$/g' \
+		| sed -E 's/(\\\.erl\\\:)/\1[[:digit:]]+:/g' \
```
Comment (javajolt) [inline tools.mk]: I was afraid this would not work on the Mac, but it seems to.

File: tools.mk [hunk 2]
```diff
@@ -17,6 +45,13 @@ test: compile
 	@echo ""
 	${REBAR} eunit skip_deps=true
 
+upload-docs: docs
```
Comment (javajolt) [inline tools.mk]: Not going to test this part, but I'll take your word for it

File: tools.mk [hunk 3]
```diff
@@ -1,7 +1,35 @@
...
+#  software distributed under the License is distributed on an
+#  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#  KIND, either express or implied.  See the License for the
+#  specific language governing permissions and limitations
+#  under the License.
+#
+#  -------------------------------------------------------------------
+
+#  -------------------------------------------------------------------
+#  NOTE: This file is is from https://github.com/basho/tools.mk.
```
Comment (javajolt) [inline tools.mk]: Probably want some of these changes over in `https://github.com/basho/tools.mk`, too

**Comment by javajolt** [comment]
Tests pass on OTP 18.3.3

**Comment by javajolt** [comment]
:+1: ce04638

File: tools.mk (hunk 1 above)
Comment (lukebakken) [inline tools.mk]: `[[:digit:]]` is the POSIX standard, while `\d` is not.

File: tools.mk (hunk 2 above)
Comment (lukebakken) [inline tools.mk]: Yeah, I just updated to the latest `tools.mk` to get the dialyzer regex fix.

**Comment by javajolt** [comment]
@borshop merge

**Pull request merged**